### PR TITLE
Enabled diagnostic reporting by default

### PR DESCRIPTION
**Pull request by nullhook**
Resolves https://github.com/brave/brave-browser/issues/27126
Resolves https://github.com/brave/brave-browser/issues/27253

Security review: https://github.com/brave/security/issues/1132

We enable diagnostic reporting by default on the new Welcome UI

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by fmarier — CHANGES_REQUESTED** [review]
I'm a bit confused. The PR is for changing the *P3A* default, but the linked issue is for changing the *crash reporting* default.

Which setting is this PR meant to change?

**Comment by nullhook** [comment]
Good catch, looks like the callbacks were swapped. Just to confirm, the first checkbox handles metrics reporting, and the second checkbox handles P3A.

**Comment by fmarier** [comment]
I was confused about this initially, but @nullhook cleared it up for me in a Slack DM:

"metrics reporting" refers to automatic crash reporting

This PR does two things:
- It corrects the naming of the checkboxes in the code (the crash report setting was controlled by the p3a variable and vice-versa) -- no user-visible changes.
- It changes the default for crash reporting to ON by default (i.e. opt out) -- covered by the [linked security review](https://github.com/brave/security/issues/1132).

**Review by fmarier — CHANGES_REQUESTED** [review]
The part of the changes which corrects the naming of the variables looks great.

I'll put my review on hold because of the second part of this change (changing the crash reporting default) until the [security review](https://github.com/brave/security/issues/1132) is complete.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.48.57 Chromium: 108.0.5359.128 (Official Build) nightly (64-bit)
-- | --
Revision | 1cd27afdb8e5d057070c0961e04c490d2aca1aa0-refs/branch-heads/5359@{#1185}
OS | Windows 11 Version 22H2 (Build 22621.963)
```
### `Test Case #1` - https://github.com/brave/brave-browser/issues/27126

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/27126#issue-1473263530, ensured that `Send diagnostic reports if Brave crashes or freezes` is set is enabled (opt-out) as per https://github.com/brave/brave-core/pull/16208#issuecomment-1338292122. Also ensured that it appeared enabled via `brave://settings/privacy`.

**`Example`** | **`Example`**
----------------|---------------
image | image

### `Test Case #2` - https://github.com/brave/brave-browser/issues/27253

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/27253#issue-1485522653, ensured that `Allow privacy-preserving product analytics (P3A)` is enabled via `brave://settings/privacy` as per the following:

**`Example`** | **`Example`**
----------------|---------------
image | image

**Comment by kjozwiak** [comment]
Created https://github.com/brave/brave-browser/issues/27393 while running through the above.

File: components/brave_welcome_ui/components/help-improve/index.tsx
Comment (bridiver) [inline components/brave_welcome_ui/components/help-improve/index.tsx:37]: this old version was wrong because it should be true for nightly and beta, but the new version is also implemented incorrectly. If we want to default to true, we should change the default pref value, not the UI

**Pull request merged**
